torch/export/exported_program.py

```diff
@@ -250,6 +280,7 @@ def _override_composite_implicit_decomp(cia_ops_to_callable, safe=True):
             op.py_kernels.clear()
             op.py_kernels.update(saved_tables[op])
             op._dispatch_cache.clear()
+            _deregister_op_impl(op)
```

Comment:
this seems wrong: above, you only registered a fake rule for ops that didn't have an existing fake rule. But here, you're *unconditionally** deregistering fake rules for all CIA ops (even if they had an existing fake rule prior to running export).

You probably want to keep a list of which ops you actually registered fake rules for, and only deregister those?